internal/wasm/store.go

```diff
@@ -157,13 +156,7 @@ type (
 const maximumFunctionTypes = 1 << 27
 
 // addSections adds section elements to the ModuleInstance
-func (m *ModuleInstance) addSections(module *Module, importedGlobals, globals []*GlobalInstance, tables []*TableInstance, memory, importedMemory *MemoryInstance,
-	types []*FunctionType,
-) {
-	m.TypeIDIndex = make(map[string]FunctionTypeID, len(types))
```

Comment:
this allocation happened every time we instantiate the module, and now it's gone!